packages/peer-exchange/src/waku_peer_exchange_discovery.ts

```diff
@@ -136,13 +141,22 @@ export class PeerExchangeDiscovery
             continue;
           }
 
-          // check if peer is already in peerStore
-          const existingPeer = await this.components.peerStore.get(peerId);
-          if (existingPeer) {
-            log("peer already in peerStore");
+          // weird hack as peerStore.get() throws an error if peer is not in peerStore
+          try {
+            // check if peer is already in peerStore
+            // continue if already exists
+            await this.components.peerStore.get(peerId);
```

Comment:
```suggestion
            await this.components.peerStore.has(peerId);
```

Use the `has` API isntead and you can remove the try/catch block

https://github.com/libp2p/js-libp2p-interfaces/blob/817701a6aecf80ed659659475d306e172e209bb4/packages/interface-peer-store/src/index.ts#L494

Reply:
good spot!